src/utils/user.js

```diff
@@ -1,17 +1,33 @@
+import { Base64 } from "js-base64";
 import asyncStore from "./asyncstorage";
+import authServices from "services/auth";
 
-async function checkIfLoggedIn() {
+async function checkForUserLogin() {
+  let user;
   try {
-    const user = JSON.parse(await asyncStore.retrieve("user"));
-    if (user && user.expireAt && user.expireAt > Date.now()) {
-      return user;
+    user = JSON.parse(await asyncStore.retrieve("user"));
+    if (!user) return null;
+    if (user instanceof Error) throw user;
+    if (user && !user.token) throw new Error("No user login token.");
+
+    // Check if header has valid info
+    const tokenHeader = JSON.parse(Base64.decode(user.token.split(".")[0]));
+    if (!tokenHeader.kid) throw new Error("Missing token key id.");
+    if (tokenHeader.kid !== user.username) {
+      throw new Error("Login token key id did not match username.");
     }
+
+    if (user) return user;
+
+    // if no user and no errors, remove stored user info from local storage
+    // to force new login bc that's shady:
     if (user) await asyncStore.remove("user");
```

Comment:
Is this something that could actually happen? If so, we could just throw here, since that's what the rest of the error handling does.

Reply:
This is a failsafe-- I wanted to catch anything that the user might have done to mess with the device storage in case the user is an adversary, but I also wanted to prevent any issues with logins from making it impossible for a user to use the app. In this case, if login tokens are messed up, a user could just try to log in again and get a fresh one.

Reply:
Alright makes sense!